### PR TITLE
Added tests for deprecated components in THM

### PR DESCRIPTION
**Pull request by licharlot**
Refs #20383

### CODE REVIEW

File: modules/thermal_hydraulics/test/tests/components/supersonic_inlet/tests [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Tests]
+  [not_implemented]
+    type = 'RunException'
+    input = 'err.i'
+    expect_err = "This component is not implemented for single-phase flow."
+    max_parallel = 1
+    ad_indexing_type = 'global'
+    design = 'SupersonicInlet.md'
+    requirement = 'The system shall throw an error if the SupersonicInlet component is used.'
```
Comment (andrsd) [inline modules/thermal_hydraulics/test/tests/components/supersonic_inlet/tests:9]: "The system shall report an error..."

File: modules/thermal_hydraulics/test/tests/components/supersonic_inlet/err.i [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  start_time = 0.0
+  end_time = 1.0
+  dt = 0.01
+  abort_on_solve_fail = true
+[]
+
+[Outputs]
+  exodus = true
+[]
```
Comment (andrsd) [inline modules/thermal_hydraulics/test/tests/components/supersonic_inlet/err.i:79]: In error checking tests, we don't have to setup any outputs like this.

File: modules/thermal_hydraulics/test/tests/components/deprecated/tests [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    expect_err = "Deprecated component. Use SolidWall1Phase or SolidWall2Phase instead."
+    max_parallel = 1
+    ad_indexing_type = 'global'
+    design = 'SolidWall.md'
+    requirement = 'The system shall throw an error if the SolidWall component is used.'
+  []
+  [junction_one_to_one]
+    type = 'RunException'
+    input = 'junction_one_to_one.i'
+    expect_err = "Depreacted component. Use JunctionOneToOne1Phase or JunctionOneToOne2Phase instead."
```
Comment (andrsd) [inline modules/thermal_hydraulics/test/tests/components/deprecated/tests:33]: "Depreacted" -> "Deprecated..."

Comment (andrsd) [reply modules/thermal_hydraulics/test/tests/components/deprecated/tests:33]: And, we will need to patch the error message in JunctionOneToOne.

File: modules/thermal_hydraulics/test/tests/components/deprecated/tests [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    ad_indexing_type = 'global'
+    design = 'HeatGeneration.md'
+    requirement = 'The system shall throw an error if the HeatGeneration component is used.'
+  []
+  [heat_source_volumetric]
+    type = 'RunException'
+    input = 'heat_source_volumetric.i'
+    expect_err = "Deprecated component. Use HeatSourceVolumetric1Phase or HeatSourceVolumetric2Phase instead."
+    max_parallel = 1
+    ad_indexing_type = 'global'
```
Comment (andrsd) [inline modules/thermal_hydraulics/test/tests/components/deprecated/tests:53]: Do we still need this? I thought that was the default. If that is so, then remove it...

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/thermal_hydraulics/test/tests/components/deprecated/tests (hunk 3 above)
Comment (licharlot) [reply modules/thermal_hydraulics/test/tests/components/deprecated/tests:33]: Good catch...

File: modules/thermal_hydraulics/test/tests/components/deprecated/tests (hunk 4 above)
Comment (licharlot) [reply modules/thermal_hydraulics/test/tests/components/deprecated/tests:53]: I think we do. Otherwise the nonsparse AD tests fail.

Comment (andrsd) [reply modules/thermal_hydraulics/test/tests/components/deprecated/tests:53]: Right. More testing on MOOSE PR... I need to remember that.

**Review by andrsd — APPROVED** [review]

**Pull request merged**